cekit/generator/docker.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import subprocess
+import yaml
+import os
+
+from cekit.errors import CekitError
+from cekit.generator.base import Generator
+from cekit.tools import cfg
+
+logger = logging.getLogger('cekit')
+
+
+class DockerGenerator(Generator):
+
+    def _prepare_repository_odcs_pulp(self, repo):
+        """Create pulp content set in ODCS and returns its url
+
+        Args:
+          repo - repository object to generate ODCS pulp for"""
+        try:
+            # idealy this will be API for ODCS, but there is no python3 package for ODCS
+            cmd = ['odcs']
+
+            if cfg.get('ODCS', {}).get('redhat'):
+                cmd.append('--redhat')
+            cmd.extend(['create', 'pulp', repo['odcs']['pulp']])
+
+            logger.debug("Creating ODCS content set via '%s'" % cmd)
+
+            output = subprocess.check_output(cmd)
+            normalized_output = '\n'.join(output.replace(" u'", " '")
+                                          .replace(' u"', ' "')
+                                          .split('\n')[1:])
+
+            odcs_result = yaml.safe_load(normalized_output)
+
+            if odcs_result['state'] == 4:
+                raise CekitError("Cannot create content set: '%s'"
```

Comment:
Isn't better to check it other way around? Is state is different from success, raise.

Reply:
Good catch again..